File: Core.Tests/EditDistanceTests.fs
namespace Core

open System
open Stringier
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type EditDistanceTests() =

    [<DataTestMethod>]
    [<DataRow(1, "ram", "rám")>]
    [<DataRow(0, "\u0072\u00E1\u006D", "\u0072\u0061\u0301\u006D")>]
    [<DataRow(1, "ram", "rom")>]
    [<DataRow(2, "ram", "rob")>]
    member _.``Hamming Distance - grapheme wise``(exp:int32, src:string, oth:string) = Assert.AreEqual(exp, hamming src oth)

    [<DataTestMethod>]
    [<DataRow(1, "ram", "rom")>]
    [<DataRow(2, "ram", "rob")>]
    member _.``Hamming Distance - character wise``(exp:int32, src:string, oth:string) = Assert.AreEqual(exp, hamming (src.ToCharArray()) (oth.ToCharArray()))

    [<DataTestMethod>]
    [<DataRow(1, "ram", "rom")>]
    [<DataRow(2, "ram", "rob")>]
    [<DataRow(3, "ram", "random")>]
    [<DataRow(2, "flaw", "lawn")>]
    member _.``Levenshtein Distance``(exp:int32, src:string, oth:string) = Assert.AreEqual(exp, levenshtein src oth)

File: Tests.CSharp/Stringier/EditDistance_Tests.cs
using System;
using Stringier;
using Xunit;

namespace Stringier {
	public class EditDistanceTests {
		[Theory]
		[InlineData("ram", "ram", 0)]
		[InlineData("ram", "rom", 1)]
		[InlineData("ram", "rob", 2)]
		[InlineData("ram", "rma", 2)]
		public void Hamming_String_String(String source, String other, Int32 expected) => Assert.Equal(expected, EditDistance.Hamming(source, other));

		[Theory]
		[InlineData("ram", "rom", 1)]
		[InlineData("\u0072\u0061\u006D", "\u0072\u0061\u0301\u006D", 1)]
		[InlineData("\u0072\u00E1\u006D", "\u0072\u0061\u0301\u006D", 2)]
		[InlineData("ram", "rob", 2)]
		[InlineData("ram", "random", 3)]
		[InlineData("flaw", "lawn", 2)]
		public void Levenshtein_String_String(String source, String other, Int32 expected) => Assert.Equal(expected, EditDistance.Levenshtein(source, other));
	}
}
